Source/SlimDetours/SlimDetours.h

```diff
@@ -31,6 +31,19 @@ HRESULT
 NTAPI
 SlimDetoursTransactionBegin(VOID);
 
+typedef struct _DETOUR_TRANSACTION_OPTIONS
+{
+    ULONG cbSize;

```

Comment:
`cbSize` is assigned but never used, this can be removed or add a check before access the following members.

Reply:
It's unused now, but the idea is to provide forward compatibility. Let's say we release this as v1.1, and at some point add a new field and call it v1.2. In v1.2, we'd access the newly added field only if `cbSize` is large enough.

Then, programs compiled for either v1.1 or v1.2 would work with DLLs of either SlimDetours v1.1 or v1.2:
* SlimDetours v1.1 will ignore the extra field for the v1.2 struct.
* SlimDetours v1.2 will use the default value for the extra field for the v1.1 struct.

`fSuspendThreads` exists in any version of the struct, so there's no need to check `cbSize` for it.

If such compatibility isn't important, I can just remove it.

Reply:
This compatibility is important for DLL exports like Windows APIs. Usually a hook library is linked staticly, there are very few such compatibility issues.

But if it is considered for future compilation to DLL, this member is useful. Both of them look good to me, feel free to choose either of them. I just report a "xxx is assigned but never used" like a compiler :D

Reply:
I think it's fine as it is now, we can add it in the future if needed.